src/H5FDsubfiling/H5subfiling_common.c

```diff
@@ -188,23 +187,27 @@ H5_get_subfiling_object(int64_t object_id)
             assert((size_t)obj_index < sf_context_cache_size);
         }
 
-        /*
-         * Since this cache currently just keeps all entries until
-         * application exit, context entry indices should just be
-         * consecutive
-         */
```

Comment:
After #4159, this cache invariant is no longer true in the case of multiple file opens and/or files closed out of order.